### ISSUE TITLE
[tool_crash] NoSuchMethodError: NoSuchMethodError: The method 'matchAsPrefix' was called on null.Receiver: nullTried calling: matchAsPrefix("ERROR: FormatException: Could not find an option named \"bytecode-options\".", 0)

### ISSUE THREAD

**Issue by loongliu** [flutter/flutter#72972]
## Command
```
flutter run --local-engine-src-path /Users/****/Develop/flutter/engine/src --local-engine=ios_debug_sim_unopt
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
NoSuchMethodError: NoSuchMethodError: The method 'matchAsPrefix' was called on null.
Receiver: null
Tried calling: matchAsPrefix("ERROR: FormatException: Could not find an option named \"bytecode-options\".", 0)
```
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _StringBase.startsWith (dart:core-patch/string_patch.dart:335:20)
#2      StdoutHandler.handler (package:flutter_tools/src/compile.dart:127:17)
#3      _rootRunUnary (dart:async/zone.dart:1198:47)
#4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
#5      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1005:7)
#6      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:357:11)
#7      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:285:7)
#8      _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:69:11)
#9      _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#10     _StringAdapterSink.add (dart:convert/string_conversion.dart:238:11)
#11     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:155:13)
#12     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:130:5)
#13     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:163:5)
#14     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:121:24)
#15     _rootRunUnary (dart:async/zone.dart:1198:47)
#16     _CustomZone

**Comment by iapicca** [comment]
@loongliu 
the error `NoSuchMethodError: NoSuchMethodError: The method 'matchAsPrefix' was called on null.`
might be caused by your implementation rather than flutter itself
could you provide `your flutter doctor -v` and a minimal reproducible code sample

**Comment by loongliu** [comment]
@iapicca Thanks for the response.

```
➜  ~ flutter doctor -v
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.5 19F101 darwin-x64, locale en-CN)
    • Flutter version 1.22.5 at /Users/xxxx/Develop/flutter/flutter
    • Framework revision 7891006299 (2 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4


[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/xxxx/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/xxxx/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[!] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    ! CocoaPods 1.7.0 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin
        usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.0

[✓] Connected device (1 available)
    • iPhone 6 (mobile) • 8AD721E2-D773-4076-BA86-22FDDF7A773B • ios • com.apple.CoreSimulator.SimRuntime.iOS-10-3
      (simulator)

! Doctor found issues in 3 categories.
```


the dart code is a flutter example without any dart code change.

**Comment by loongliu** [comment]
this is a flutter tool crash

**Comment by jmagman** [comment]
`bytecode-options` was removed from the VM in https://github.com/dart-lang/sdk/commit/1615c04c63f50f2788dfa1eb6308252aa10555ef in dart 2.11.0-252.0.dev and stopped being passed along in the Flutter tool in #67755, which is in 1.26.0-1.0.pre, currently on available on the [dev channel](https://flutter.dev/docs/development/tools/sdk/releases?tab=macos).  However you are running Flutter tool stable 1.22.5.  

If you're building a local engine based off the top of tree of the engine you need to be on the Flutter master channel to pick up all the lock-step changes made between dart and the tool.

In other words, to use the local engine you built, you need to be on Flutter version 1.26.0-1.0.pre or newer.
